airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -214,27 +211,40 @@ private static Container getMain(final String image,
     return containerBuilder.build();
   }
 
-  private static void copyFilesToKubeConfigVolume(final ApiClient officialClient,
-                                                  final String podName,
+  private static void copyFilesToKubeConfigVolume(final String podName,
                                                   final String namespace,
                                                   final Map<String, String> files) {
     final List<Map.Entry<String, String>> fileEntries = new ArrayList<>(files.entrySet());
 
     // copy this file last to indicate that the copy has completed
     fileEntries.add(new AbstractMap.SimpleEntry<>(SUCCESS_FILE_NAME, ""));
 
+    Path tmpFile = null;
     for (final Map.Entry<String, String> file : fileEntries) {
       try {
+        tmpFile = Path.of(IOs.writeFileToRandomTmpDir(file.getKey(), file.getValue()));
+
         LOGGER.info("Uploading file: " + file.getKey());
-        final var contents = file.getValue().getBytes(StandardCharsets.UTF_8);
         final var containerPath = Path.of(CONFIG_DIR + "/" + file.getKey());
 
-        // fabric8 kube client upload doesn't work on gke:
-        // https://github.com/fabric8io/kubernetes-client/issues/2217
-        final Copy copy = new Copy(officialClient);
-        copy.copyFileToPod(namespace, podName, INIT_CONTAINER_NAME, contents, containerPath);
-      } catch (final IOException | ApiException e) {
+        // using kubectl cp directly here, because both fabric and the official kube client APIs both have
+        // several issues with copying files. See https://github.com/airbytehq/airbyte/issues/8643 for
+        // detauls.
+        final String command = String.format("kubectl cp %s %s/%s:%s -c %s", tmpFile, namespace, podName, containerPath, INIT_CONTAINER_NAME);
+        LOGGER.info(command);
+
+        final Process proc = Runtime.getRuntime().exec(command);
+        LOGGER.info("Waiting for kubectl cp to complete");
+        proc.waitFor();
```

Comment:
@jrhizor I pushed some changes to make this handle the exit code of the process. One note there: I had to increase the `sleep` time of the `init` container entrypoint from 0.00001 to 0.1 seconds: https://github.com/airbytehq/airbyte/pull/8880/files#diff-be3c5f5544310ca55f284848493a7afe72533bd82a737a044d3f58e9c82d71abR157
Without this change, the final `kubectl cp` command to create the `FINISHED_UPLOADING` file would exit with code 137, because the creation of that file would cause the init container to shut down, causing the `kubectl cp` command to be SIGKILLed. 0.1 seconds of sleeping after init sees the file and before it exits seems to give `kubectl cp` enough time to exit gracefully, and I don't think anyone will notice the difference between 0.1s and 0.00001s.

Reply:
I'm a it worried about using 0.1 second to retrieve the process's exit code. Can we check if the init pod succeeded if we get a sigkill for the success indicator file?

Reply:
@jrhizor I just tried that out and it looks like at the time that `kubectl cp` exits with code 137 for the success indicator file, the init container is still running, so I don't think that will work. My guess is that the processes in the container get closed before the container actually terminates

Reply:
I can try adding logic to wait until the init container terminates before doing the check to see if it succeeded

Reply:
@jrhizor that worked, I have pushed the changes and tested them locally

Reply:
nice